business/suggestionService.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const { get, set, sortBy, filter, concat, last, isEmpty } = require('lodash')
+
+class SuggestionService {
+  constructor(definitionService, definitionStore) {
+    this.definitionService = definitionService
+    this.definitionStore = definitionStore
+  }
+
+  /**
+   * Get suggestions for the given coordinates.
+   *
+   * @param {EntityCoordinates} coordinates - The entity for which we are looking for suggestions
+   * @returns {Suggestion} A set of suggested fixes for the definition at the given coordinates.
+   * `null` is returned if no such coordinates are found.
+   */
+  async get(coordinates) {
+    const related = await this._getRelatedDefinitions(coordinates)
+    if (!related) return null
+    const baseSuggestion = this._createBaseSuggestion(coordinates)
+    // create a struct with the strings in it and iterate instead
+    const promises = [
+      await this._collectSuggestionsForField(related, baseSuggestion, 'licensed.declared'),
+      await this._collectSuggestionsForField(related, baseSuggestion, 'described.sourceLocation'),
+      await this._collectSuggestionsForField(related, baseSuggestion, 'described.releaseDate')
+    ]
+    const result = await Promise.all(promises)
+    return last(result.filter(res => !isEmpty(res)))
```

Comment:
Full word names please

Reply:
Given another comment, i'm gonna remove this method and return `baseSuggestion`